tensorboard/plugins/histogram/metadata.py

```diff
@@ -53,6 +52,19 @@ def parse_plugin_metadata(content):
       corresponding to the histogram plugin.
 
   Returns:
-    A `HistogramMetadata` instance.
+    A `HistogramPluginData` protobuf object.
   """
-  return HistogramMetadata(**json.loads(content))
+  if content == '{}':
+    # Old-style JSON format. Equivalent to an all-default proto.
+    return plugin_data_pb2.HistogramPluginData()
+  else:
+    result = plugin_data_pb2.HistogramPluginData()
+    result.ParseFromString(tf.compat.as_bytes(content))
```

Comment:
Unfortunately, [the `PluginData` proto declares `string content = 2`, not `bytes content = 2`](https://github.com/tensorflow/tensorflow/blob/3686ef0d51047d2806df3e2ff6c1aac727456c1d/tensorflow/core/framework/summary.proto#L45). As a consequence, in Python3:
```py
>>> import tensorflow as tf
>>> content = b'123'
>>> type(content)
<class 'bytes'>
>>> pd0 = tf.SummaryMetadata.PluginData(content=content)
>>> pd0.content
'123'
>>> type(pd0.content)
<class 'str'>
```